openhtf/core/__init__.py

```diff
@@ -114,6 +114,10 @@ def _thread_proc(self):
       # Any access to self._exit_stack must be done while holding this lock.
       with self._lock:
         self._exit_stack = exit_stack
+        # We need to exit the PhaseExecutor before we tear down the plugs
+        # that a phase may be executing, so we add them in reverse order.
+        exit_stack.callback(self.test_state.plug_manager.tear_down_plugs)
+        exit_stack.callback(executor.stop)
```

Comment:
`PhaseExecutor.stop()` will only kill its current phase, if any. If it's called right after this block, there will be no phases so nothing happens. `PlugManager.tear_down_plugs()` won't do anything when it hasn't initialized any plugs, and will only tear down those that it has initialized. There is no lock though, so if you hit at the right time, there's a possible race condition that we'll need to address some other way. I propose that we move `self.kill()` above `self._exit_stack.close()` in `stop()` above (in another PR).